src/main/java/com/hivemq/extensions/handler/PluginInitializerHandler.java

```diff
@@ -121,6 +122,7 @@ private void fireInitialize(
 
         //No initializer set through any extension
         if (pluginInitializerMap.isEmpty() && msg != null) {
+            removeHandler(ctx);
```

Comment:
I think the handler can directly be removed after calling fireInitialize in line 110.
Then the removeHandler helper method is also not necessary, simply call `ctx.pipeline().remove(this)`